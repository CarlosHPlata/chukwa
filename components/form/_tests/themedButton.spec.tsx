import { render, fireEvent } from "@testing-library/react-native";
import ThemedButton from "../themedButton";

describe("ThemedButton", () => {
  it("should render the button correcly", () => {
    const tree = render(
      <ThemedButton label="Click Me" onPress={() => {}} />,
    );

    expect(tree).toMatchSnapshot();
  });

  it("should handle press events", () => {
    const mockOnPress = jest.fn();
    const renderTree = render(
      <ThemedButton label="Click Me" onPress={mockOnPress} />,
    );

    const button = renderTree.getByTestId("themed-button:Pressable");
    fireEvent.press(button);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it("should display the correct label", () => {
    const label = "Click Me";
    const renderTree = render(
      <ThemedButton label={label} onPress={() => {}} />,
    );

    const buttonLabel = renderTree.getByTestId("themed-button:Label");
    expect(buttonLabel.props.children).toBe(label);
  });
});
