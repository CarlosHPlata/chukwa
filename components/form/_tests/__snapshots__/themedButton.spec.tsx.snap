// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedButton should render the button correcly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#25292e",
          "borderRadius": 10,
          "flexGrow": 1,
          "marginHorizontal": 30,
          "paddingVertical": 10,
        },
      ]
    }
    testID="themed-button:Pressable"
  >
    <Text
      style={
        {
          "color": "#fff",
        }
      }
      testID="themed-button:Label"
    >
      Click Me
    </Text>
  </View>
</View>
`;
